src/runtime/fetcher.js

```diff
@@ -60,4 +68,19 @@ export class XhrFetcher {
   fetch(url, init) {
     return this.xhr_.fetch(url, init);
   }
+
+  /** @override */
+  sendBeacon(url, data) {
+    url = addQueryParam(url, 'f.req', serializeMessageForUrl(data));
+    if (navigator.sendBeacon) {
+      navigator.sendBeacon(url);
```

Comment:
Why not send the `data` as the second parameter to `sendBeacon`, which includes it in the POST body instead of the URL, to avoid serialization and potentially running out of space in the URL? https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon

Reply:
I spent 2 days trying every possible serialization format using the second parameter.  BOQ rejected them all.  I had to manually hack at BOQ until I got it to accept one of the serialization formats I randomly guessed (this odd query string thing is the only one that worked).  

I was not able to find any assistance in YAQS or moma defining the format it was expecting (they just tell you to use JSPB which we can't use).

I have at this point been able to confirm that the server is saving the proto I send to clearcut so deserialization is working right now.

Reply:
The server should be able to handle binary protobuf wire format for the request body. Not sure if our lightweight js protobufs have a way to serialize to binary format.

Reply:
Do you have any idea what the binary format looks like?  Because I couldn't find anything defining the proper format, I just guessed until I found something that worked.

Reply:
The blob method you indicated I should use is rejected by Chrome currently because of a security issue.  The URL size in scenic is under 1000 - I do agree this is a slightly concern of the 2048 character URL limit though I think nothing should actually hit that.

Reply:
FormData is more appropriate, because of the f.req param.

https://developer.mozilla.org/en-US/docs/Web/API/FormData

Reply:
It sends the wrong content-type.  Blob is the one we need but Chrome has it disabled.  I am looking into work arounds.